kratos/processes/calculate_embedded_nodal_variable_from_skin_process.h

```diff
@@ -0,0 +1,729 @@
+inline void EmbeddedNodalVariableFromSkinTypeHelperClass<array_1d<double, 3>>::AddUnknownVariableDofs(ModelPart &rModelPart)
+{
+    VariableUtils().AddDof(NODAL_VAUX_X, rModelPart);
+    VariableUtils().AddDof(NODAL_VAUX_Y, rModelPart);
+    VariableUtils().AddDof(NODAL_VAUX_Z, rModelPart);
+}
+
+template <class TVarType, class TSparseSpace, class TDenseSpace, class TLinearSolver>
+class KRATOS_API(KRATOS_CORE) CalculateEmbeddedNodalVariableFromSkinProcess : public Process
+{
+public:
+
+    ///@name Type Definitions
+    ///@{
+
+    typedef typename Element::NodeType::Pointer NodePointerType;
+    typedef typename Scheme<TSparseSpace,TDenseSpace>::Pointer SchemePointerType;
+    typedef typename BuilderAndSolver<TSparseSpace,TDenseSpace,TLinearSolver>::Pointer BuilderSolverPointerType;
+    typedef typename SolvingStrategy<TSparseSpace, TDenseSpace, TLinearSolver>::UniquePointer SolvingStrategyPointerType;
+    typedef typename FindIntersectedGeometricalObjectsProcess::Pointer FindIntersectedGeometricalObjectsProcessPointerType;
+
+    struct Hash
+    {
+        std::size_t operator()(const std::pair<unsigned int,bool>& k) const
+        {
+            std::size_t h1 = std::hash<unsigned int>()(std::get<0>(k));
+            std::size_t h2 = std::hash<bool>()(std::get<1>(k));
+            return h1 ^ (h2 << 1);
+        }
+    };
+
+    struct KeyEqual
+    {
+        bool operator()(const std::pair<std::size_t, std::size_t>& lhs, const std::pair<std::size_t, std::size_t>& rhs) const
+        {
+            return ((std::get<0>(lhs) == std::get<0>(rhs)) && (std::get<1>(lhs) == std::get<1>(rhs)));
+        }
+    };
+
+    typedef std::unordered_map<std::pair<std::size_t, std::size_t>, std::size_t, Hash, KeyEqual> EdgesMapType;
+
+    ///@}
+    ///@name Pointer Definitions
+
+    /// Pointer definition of CalculateEmbeddedNodalVariableFromSkinProcess
+    KRATOS_CLASS_POINTER_DEFINITION(CalculateEmbeddedNodalVariableFromSkinProcess);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    CalculateEmbeddedNodalVariableFromSkinProcess(
+        ModelPart &rBaseModelPart,
+        ModelPart &rSkinModelPart,
+        const Variable<TVarType> &rSkinVariable,
+        const Variable<TVarType> &rEmbeddedNodalVariable,
+        const std::string LevelSetType = "continuous",
+        const std::string AuxPartName = "IntersectedElementsModelPart")
+        : mLevelSetType(LevelSetType),
+          mAuxModelPartName(AuxPartName),
+          mrBaseModelPart(rBaseModelPart),
+          mrSkinModelPart(rSkinModelPart),
+          mrSkinVariable(rSkinVariable),
+          mrEmbeddedNodalVariable(rEmbeddedNodalVariable)
+    {
+        KRATOS_TRY
+
+        // Check that there is at least one element and node in the model
+        KRATOS_ERROR_IF(mrBaseModelPart.NumberOfNodes() == 0) << "The model part has no nodes." << std::endl;
+        KRATOS_ERROR_IF(mrBaseModelPart.NumberOfElements() == 0) << "The model Part has no elements." << std::endl;
```

Comment:
should this work in MPI too? if yes then this is not correct, the global number has to be checked

Reply:
In the future yes... Currently it is impossible because of the construction of the model part of interest. In any case I tried to take the MPI migration into consideration so I'll change it in any case.

Reply:
Done. Checking the summation of all the local meshes entities.